Makefile

```diff
@@ -21,82 +21,114 @@ endif
 
 all: manager
 
-# Create a local docker registry, start kind cluster, and install Orkestra
+## Create kind cluster with local registry and prepopulate the local registry.
+setup-dev-env: kind-create prepopulate-kind-registry
+
+## Deploy the Orkestra helm chart (values-ci.yaml) with Orkestra controller disabled.
 dev:
-	make kind-create
 	helm upgrade --install orkestra chart/orkestra --wait --atomic -n orkestra --create-namespace --values ${CI_VALUES}
 
+## Deploy the Orkestra helm chart (values-ci.yaml) with Orkestra controller enabled (in debug mode).
 debug: dev
 	go run main.go --debug --log-level ${DEBUG_LEVEL}
 
+## Cleanup any previous Orkestra helm chart installation.
+clean:
+	helm delete orkestra -n orkestra 2>&1 || true
+	@rm -rf $(BIN_DIR)
+	@echo "> 👍 Done\n"
+
+## Cleanup any Orkestra installation and the kind cluster with registry.
+clean-all: kind-delete
+	@rm -rf $(BIN_DIR)
+
+## Run tests.
+test: install
+	go test -v ./... -coverprofile coverage.txt -timeout 35m
+
+## Run tests using Ginkgo.
 ginkgo-test: install
 	go get github.com/onsi/ginkgo/ginkgo
 	ginkgo ./... -cover -coverprofile coverage.txt
 
-# Run tests
-test: install
-	go test -v ./... -coverprofile coverage.txt -timeout 35m
+## Prepare code for PR.
+prepare-for-pr: vet fmt api-docs
+	@echo "\n> ❗️ Remember to run the tests"
 
-# Build manager binary
+## Build manager binary.
 manager: generate fmt vet
 	go build -o bin/manager main.go
 
-# Run against the configured Kubernetes cluster in ~/.kube/config
+## Run against the configured Kubernetes cluster in ~/.kube/config.
 run: generate fmt vet manifests
 	go run ./main.go
 
-# Install CRDs into a cluster
+## Install CRDs into a cluster.
 install: manifests
 	kustomize build config/crd | kubectl apply -f -
 
-# Uninstall CRDs from a cluster
+## Uninstall CRDs from a cluster.
 uninstall: manifests
 	kustomize build config/crd | kubectl delete -f -
 
-# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+# #Deploy controller in the configured Kubernetes cluster in ~/.kube/config.
 deploy: manifests
 	cd config/manager && kustomize edit set image controller=${IMG}
 	kustomize build config/default | kubectl apply -f -
 
-# Generate manifests e.g. CRD, RBAC etc.
+## Generate code.
+generate: controller-gen
```

Comment:
Why do we need a second generate file? This strikes me as redundant

Reply:
@nitishm Can you look into this.

Reply:
Move controller-gen target calls to generate. Jonathan is right. This is redundant and generate is more intuitive in my opinion